buildshiprun/handler.go

```diff
@@ -269,89 +217,68 @@ func enableStatusReporting() bool {
+	_, reportErr := status.Report(gatewayURL)
+	if reportErr != nil {
+		log.Printf("failed to report status, error: %s", reportErr.Error())
+	}
+}
 
-	log.Printf("Status: %s, GitHub AppID: %d, Repo: %s, Owner: %s", status, event.installationID, event.repository, event.owner)
+func getImageName(repositoryURL, pushRepositoryURL, imageName string) string {
 
-	token, tokenErr := auth.MakeAccessTokenForInstallation(os.Getenv("github_app_id"), event.installationID, privateKeyPath)
+	return strings.Replace(imageName, pushRepositoryURL, repositoryURL, 1)
 
-	if tokenErr != nil {
-		fmt.Printf("failed to report status %v, error: %s\n", repoStatus, tokenErr.Error())
-		return
-	}
+	// return repositoryURL + imageName[strings.Index(imageName, "/"):]
+}
 
-	if token == "" {
-		fmt.Printf("failed to report status %v, error: authentication failed Invalid token\n", repoStatus)
-		return
-	}
+// function to build Event from Environment Variable
+func BuildEventFromEnv() (*sdk.Event, error) {
+	var err error
+	info := sdk.Event{}
 
-	client := auth.MakeClient(ctx, token)
+	info.Service = os.Getenv("Http_Service")
+	info.Owner = os.Getenv("Http_Owner")
+	info.Repository = os.Getenv("Http_Repo")
+	info.Sha = os.Getenv("Http_Sha")
+	info.URL = os.Getenv("Http_Url")
+	info.InstallationID, err = strconv.Atoi(os.Getenv("Http_Installation_id"))
+	info.Environment = GetEnv(info.Service)
+	info.Secrets = GetSecret(info.Owner, info.Service)
 
-	_, _, apiErr := client.Repositories.CreateStatus(ctx, event.owner, event.repository, event.sha, repoStatus)
-	if apiErr != nil {
-		fmt.Printf("failed to report status %v, error: %s\n", repoStatus, apiErr.Error())
-		return
-	}
+	return &info, err
 }
 
-func getPrivateKey() string {
-	// we are taking the secrets name from the env, by default it is fixed
-	// to private_key.pem.
-	// Although user can make the secret with a specific name and provide
-	// it in the stack.yaml and also specify the secret name in github.yml
-	privateKeyName := os.Getenv("private_key")
-	if privateKeyName == "" {
-		privateKeyName = defaultPrivateKeyName
+func GetEnv(service string) map[string]string {
+	envVars := make(map[string]string)
+	envStr := os.Getenv("Http_Env")
+	if len(envStr) > 0 {
+		envErr := json.Unmarshal([]byte(envStr), &envVars)
+		if envErr != nil {
+			log.Printf("error un-marshaling env-vars for function %s, %s", service, envErr)
+		}
 	}
-	privateKeyPath := "/var/openfaas/secrets/" + privateKeyName
-	return privateKeyPath
-}
-
-func buildStatus(status string, desc string, context string, url string) *github.RepoStatus {
-	return &github.RepoStatus{State: &status, TargetURL: &url, Description: &desc, Context: &context}
+	return envVars
 }
 
-func getImageName(repositoryURL, pushRepositoryURL, imageName string) string {
-
-	return strings.Replace(imageName, pushRepositoryURL, repositoryURL, 1)
-
-	// return repositoryURL + imageName[strings.Index(imageName, "/"):]
+func GetSecret(owner, service string) []string {
```

Comment:
Upper-case letter denotes a public method, should it be lower-case?

Is there a corresponding unit test for this?